sonar-text-plugin/src/main/java/org/sonar/plugins/secrets/api/SpecificationBasedCheck.java

```diff
@@ -48,29 +52,37 @@ protected SpecificationBasedCheck() {
     super();
   }
 
-  public void initialize(SpecificationLoader loader, Map<InputFileContext, List<TextRange>> reportedIssuesForCtx) {
+  /**
+   * Initialize this check by loading rule definitions
+   * @param loader SpecificationLoader that will provide specification files
+   * @param reportedIssuesForCtx a map to store reported issues to prevent duplication among different checks
+   * @param durationStatistics an instance to record performance statistics
+   */
+  public void initialize(SpecificationLoader loader, Map<URI, List<TextRange>> reportedIssuesForCtx, DurationStatistics durationStatistics) {
     this.reportedIssuesForCtx = reportedIssuesForCtx;
+    this.durationStatistics = durationStatistics;
     String rule = ruleKey.rule();
     List<Rule> rulesForKey = loader.getRulesForKey(rule);
     if (rulesForKey.isEmpty()) {
       LOG.error("Found no rule specification for rule with key: {}", rule);
     }
     this.matcher = rulesForKey.stream()
-      .map(SecretMatcher::build)
+      .map(r -> SecretMatcher.build(r, durationStatistics))
```

Comment:
```suggestion
      .map(rule -> SecretMatcher.build(rule, durationStatistics))
```

Reply:
Did some renaming around, because `ruleKey.rule()` is actually a `ruleId`